agr_literature_service/api/routers/resource_router.py

```diff
@@ -69,3 +70,20 @@ def show(curie: str,
 def show_versions(curie: str,
                   db: Session = db_session):
     return resource_crud.show_changesets(db, curie)
+
+
+@router.get('/check/patterns',
+            status_code=200,
+            )
+def show_resource_patterns():
+    return patterns_check.get_patterns()['resource']
+
+
+@router.get('/check/{curie_prefix}/{curie}',
```

Comment:
This might be a little ambiguous, because it's expecting curie as a second parameter, but it's expecting to get the after-colon part of the curie instead of the whole curie.  Maybe check_pattern in patterns_check could do the check against the right side like now, but also against the curie_prefix:right_side passed in here, and return true if either passes ?  I'm not really sure.  Maybe it should expect the full curie, or it should explicitly ask for curie_right_side (don't know what to call that).  Would also welcome other dev feedback on this.

Reply:
My reasoning was that other code has to split this up anyway to store the prefix, so the splitting will have been done already. As for the ui I presumed less typing for curators was better so no point selecting MGI as the prefix and then typing in MGI:123456. It is easy to change but as you say it's up to the curators really. Happy either way.